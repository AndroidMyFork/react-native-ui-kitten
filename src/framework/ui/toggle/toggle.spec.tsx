import React from 'react';
import {
  render,
  shallow,
  RenderAPI,
} from 'react-native-testing-library';
import {
  styled,
  StyleProvider,
  StyleProviderProps,
} from '@kitten/theme';
import {
  Toggle,
  ToggleProps,
} from './toggle.component';
import * as config from './toggle.spec.config';

const StyledComponent = styled<Toggle, ToggleProps>(Toggle);

const Mock = (props?: ToggleProps): React.ReactElement<StyleProviderProps> => (
  <StyleProvider mapping={config.mapping} theme={config.theme} styles={{}}>
    <StyledComponent {...props} />
  </StyleProvider>
);

const renderComponent = (props?: ToggleProps): RenderAPI => render(<Mock {...props}/>);

describe('@toggle: matches snapshot', () => {

  it('default', () => {
    const component = renderComponent();
    const { output } = shallow(component.getByType(Toggle));

    expect(output).toMatchSnapshot();
  });

  it('checked', () => {
    const component = renderComponent({ value: true });
    const { output } = shallow(component.getByType(Toggle));

    expect(output).toMatchSnapshot();
  });

  it('disabled', () => {
    const component = renderComponent({ disabled: true });
    const { output } = shallow(component.getByType(Toggle));

    expect(output).toMatchSnapshot();
  });

  it('checked disabled', () => {
    const component = renderComponent({ value: true, disabled: true });
    const { output } = shallow(component.getByType(Toggle));

    expect(output).toMatchSnapshot();
  });

});
