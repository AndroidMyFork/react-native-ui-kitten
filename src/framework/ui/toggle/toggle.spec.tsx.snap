// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@toggle: matches snapshot checked 1`] = `
<View
  style={
    Array [
      Object {
        "height": 32,
        "width": 52,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "position": "absolute",
        },
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 21,
          "height": 42,
          "width": 62,
        },
      ]
    }
  />
  <AnimatedComponent
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "alignSelf": "center",
          "justifyContent": "center",
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 16,
          "height": 32,
          "width": 52,
        },
        Object {
          "backgroundColor": "rgba(51, 102, 255, 1)",
        },
      ]
    }
  >
    <AnimatedComponent
      style={
        Array [
          Object {
            "alignSelf": "center",
            "position": "absolute",
          },
          Object {
            "borderRadius": 14.5,
            "height": 29,
            "width": 49,
          },
          Object {
            "backgroundColor": "rgba(51, 102, 255, 1)",
            "transform": Array [
              Object {
                "scale": 0.010000000000000009,
              },
            ],
          },
        ]
      }
    />
    <AnimatedComponent
      style={
        Array [
          Object {
            "borderRadius": 14.5,
            "borderWidth": 1.5,
            "height": 29,
            "marginHorizontal": 1.5,
            "width": 29,
          },
          Object {
            "alignSelf": "flex-end",
            "backgroundColor": "#ffffff",
            "borderColor": "rgba(51, 102, 255, 1)",
            "elevation": 5,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
            "width": 29,
          },
        ]
      }
    />
  </AnimatedComponent>
</View>
`;

exports[`@toggle: matches snapshot checked disabled 1`] = `
<View
  style={
    Array [
      Object {
        "height": 32,
        "width": 52,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <AnimatedComponent
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "alignSelf": "center",
          "justifyContent": "center",
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 16,
          "height": 32,
          "width": 52,
        },
        Object {
          "backgroundColor": "rgba(137, 146, 163, 1)",
        },
      ]
    }
  >
    <AnimatedComponent
      style={
        Array [
          Object {
            "alignSelf": "center",
            "position": "absolute",
          },
          Object {
            "borderRadius": 14.5,
            "height": 29,
            "width": 49,
          },
          Object {
            "backgroundColor": "rgba(137, 146, 163, 1)",
            "transform": Array [
              Object {
                "scale": 0.010000000000000009,
              },
            ],
          },
        ]
      }
    />
    <AnimatedComponent
      style={
        Array [
          Object {
            "borderRadius": 14.5,
            "borderWidth": 1.5,
            "height": 29,
            "marginHorizontal": 1.5,
            "width": 29,
          },
          Object {
            "alignSelf": "flex-end",
            "backgroundColor": "#ffffff",
            "borderColor": "rgba(137, 146, 163, 1)",
            "elevation": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
            "width": 29,
          },
        ]
      }
    />
  </AnimatedComponent>
</View>
`;

exports[`@toggle: matches snapshot default 1`] = `
<View
  style={
    Array [
      Object {
        "height": 32,
        "width": 52,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "position": "absolute",
        },
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 21,
          "height": 42,
          "width": 62,
        },
      ]
    }
  />
  <AnimatedComponent
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "alignSelf": "center",
          "justifyContent": "center",
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 16,
          "height": 32,
          "width": 52,
        },
        Object {
          "backgroundColor": "rgba(221, 225, 235, 1)",
        },
      ]
    }
  >
    <AnimatedComponent
      style={
        Array [
          Object {
            "alignSelf": "center",
            "position": "absolute",
          },
          Object {
            "borderRadius": 14.5,
            "height": 29,
            "width": 49,
          },
          Object {
            "backgroundColor": "rgba(221, 225, 235, 1)",
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          },
        ]
      }
    />
    <AnimatedComponent
      style={
        Array [
          Object {
            "borderRadius": 14.5,
            "borderWidth": 1.5,
            "height": 29,
            "marginHorizontal": 1.5,
            "width": 29,
          },
          Object {
            "alignSelf": "flex-start",
            "backgroundColor": "#ffffff",
            "borderColor": "rgba(221, 225, 235, 1)",
            "elevation": 5,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
            "width": 29,
          },
        ]
      }
    />
  </AnimatedComponent>
</View>
`;

exports[`@toggle: matches snapshot disabled 1`] = `
<View
  style={
    Array [
      Object {
        "height": 32,
        "width": 52,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <AnimatedComponent
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "alignSelf": "center",
          "justifyContent": "center",
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 16,
          "height": 32,
          "width": 52,
        },
        Object {
          "backgroundColor": "rgba(221, 225, 235, 1)",
        },
      ]
    }
  >
    <AnimatedComponent
      style={
        Array [
          Object {
            "alignSelf": "center",
            "position": "absolute",
          },
          Object {
            "borderRadius": 14.5,
            "height": 29,
            "width": 49,
          },
          Object {
            "backgroundColor": "rgba(221, 225, 235, 1)",
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          },
        ]
      }
    />
    <AnimatedComponent
      style={
        Array [
          Object {
            "borderRadius": 14.5,
            "borderWidth": 1.5,
            "height": 29,
            "marginHorizontal": 1.5,
            "width": 29,
          },
          Object {
            "alignSelf": "flex-start",
            "backgroundColor": "#ffffff",
            "borderColor": "rgba(221, 225, 235, 1)",
            "elevation": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
            "width": 29,
          },
        ]
      }
    />
  </AnimatedComponent>
</View>
`;
